core/src/main/resources/org/broadinstitute/dsde/rawls/liquibase/changesets/20230628_multiregional_bucket_migration.xml

```diff
@@ -0,0 +1,249 @@
+            </column>
+        </createTable>
+    </changeSet>
+    <changeSet logicalFilePath="dummy" author="mtalbott" id="create_MULTIREGIONAL_BUCKET_MIGRATION_HISTORY_FK">
+        <addForeignKeyConstraint baseColumnNames="WORKSPACE_ID"
+                                 baseTableName="MULTIREGIONAL_BUCKET_MIGRATION_HISTORY"
+                                 constraintName="FK_MULTIREGIONAL_BUCKET_MIGRATION_HISTORY_WORKSPACE"
+                                 deferrable="false"
+                                 initiallyDeferred="false"
+                                 onDelete="CASCADE"
+                                 onUpdate="NO ACTION"
+                                 referencedColumnNames="id"
+                                 referencedTableName="WORKSPACE"/>
+    </changeSet>
+    <changeSet logicalFilePath="dummy" author="mtalbott"
+               id="create_MULTIREGIONAL_BUCKET_MIGRATION_HISTORY_insertion_trigger">
+        <comment>
+            The task of migrating a multiregional bucket to a single region might fail for a number
+            of reasons, and it's important that we record these failures. An attempt at migrating
+            any particular workspace bucket is represented by an insertion into
+            MULTIREGIONAL_BUCKET_MIGRATION_HISTORY. To prevent concurrent attempts, we'll only allow
+            insertions when previous attempts have FINISHED.
+        </comment>
+        <sql endDelimiter="~">
+            CREATE TRIGGER fail_insert_if_workspace_bucket_is_being_migrated
+                BEFORE INSERT ON MULTIREGIONAL_BUCKET_MIGRATION_HISTORY
+                FOR EACH ROW
+            BEGIN
+                IF (SELECT COUNT(*) FROM MULTIREGIONAL_BUCKET_MIGRATION_HISTORY h
+                    WHERE h.WORKSPACE_ID = NEW.WORKSPACE_ID AND h.FINISHED IS NULL)
+                THEN
+                    SET @error_message = CONCAT_WS(' ', 'Workspace', NEW.WORKSPACE_ID, 'is already being migrated.');
+                SIGNAL SQLSTATE '45000' SET MESSAGE_TEXT = @error_message;
+            END IF;
+            END ~
+        </sql>
+        <rollback>
+            DROP TRIGGER IF EXISTS fail_insert_if_workspace_bucket_is_being_migrated;
+        </rollback>
+    </changeSet>
+    <changeSet logicalFilePath="dummy" author="mtalbott" id="create_MULTIREGIONAL_BUCKET_MIGRATION_STORAGE_TRANSFER_SERVICE_JOB">
+        <createTable tableName="MULTIREGIONAL_BUCKET_MIGRATION_STORAGE_TRANSFER_SERVICE_JOB"
+                     remarks="Records the job information for STS jobs used in multiregional bucket migration.">
+            <column name="id" type="BIGINT" autoIncrement="true">
+                <constraints primaryKey="true"/>
+            </column>
+            <column name="JOB_NAME"
+                    type="varchar(254)"
+                    remarks="Name of STS job.">
+                <constraints nullable="false"/>
+            </column>
+            <column name="MIGRATION_ID"
```

Comment:
This is purely for the linkage to the retry count, correct?

Reply:
This links back to the primary migration table so we can associate STS jobs with the appropriate migration. We can also link to retry count through this